model/PermissionsService.php

```diff
@@ -54,55 +56,150 @@ public function __construct(
         $this->eventManager = $eventManager;
     }
 
-    public function saveResourcePermissionsRecursive(
+    /**
+     * Returns an opaque array holding the list of actions to be performed
+     * (i.e. a permissions diff) to fulfill a user's change permissions request.
+     */
+    public function getNormalizedRequest(
```

Comment:
Why not keeping the existing methods working? Why create so many new public methods?